Sources/Lifetime.swift

```diff
@@ -39,6 +39,58 @@ public final class Lifetime {
 		self.init(ended: token.ended)
 	}
 
+	/// Create a lifetime that is an OR of `self` and `other`.
+	///
+	/// - parameters:
+	///   - other: The lifetime to be composed with.
+	///
+	/// - returns:
+	///   An OR lifetime.
+	public func or(_ other: Lifetime) -> Lifetime {
+		return Lifetime(ended: ended.zip(with: other.ended).map { _ in })
+	}
+
+	/// Create a lifetime that is an AND of `self` and `other`.
```

Comment:
> But of course these can be eliminated if notifications(forName:object:) would terminate the returned signal if object has deinitialized.

That is actually the case: https://github.com/ReactiveCocoa/ReactiveCocoa/pull/2747.